host/namespace_delete_test.go

```diff
@@ -0,0 +1,245 @@
+	s.NoError(err)
+	s.Equal(enumspb.NAMESPACE_STATE_DELETED, descResp2.GetNamespaceInfo().GetState())
+
+	namespaceExistsOp := func() error {
+		_, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+			Id: nsID,
+		})
+		var notFound *serviceerror.NamespaceNotFound
+		if errors.As(err, &notFound) {
+			return nil
+		}
+		return errors.New("namespace still exists")
+	}
+
+	namespaceExistsPolicy := backoff.NewExponentialRetryPolicy(time.Second)
+	namespaceExistsPolicy.SetBackoffCoefficient(1)
+	namespaceExistsPolicy.SetExpirationInterval(30 * time.Second)
+
+	err = backoff.ThrottleRetry(namespaceExistsOp, namespaceExistsPolicy, func(_ error) bool { return true })
+	s.NoError(err)
+}
+
+func (s *namespaceTestSuite) Test_NamespaceDelete_WithWorkflows() {
+	ctx, cancel := rpc.NewContextWithTimeoutAndVersionHeaders(10000 * time.Second)
+	defer cancel()
+
+	retention := 24 * time.Hour
+	_, err := s.frontendClient.RegisterNamespace(ctx, &workflowservice.RegisterNamespaceRequest{
+		Namespace:                        "ns_name_los_angeles",
+		Description:                      "Namespace to delete",
+		WorkflowExecutionRetentionPeriod: &retention,
+		HistoryArchivalState:             enumspb.ARCHIVAL_STATE_DISABLED,
+		VisibilityArchivalState:          enumspb.ARCHIVAL_STATE_DISABLED,
+	})
+	s.NoError(err)
+	// DescribeNamespace reads directly from database but namespace validator uses cache.
+	s.cluster.RefreshNamespaceCache()
+
+	descResp, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Namespace: "ns_name_los_angeles",
+	})
+	s.NoError(err)
+	nsID := descResp.GetNamespaceInfo().GetId()
+
+	// Start few workflow executions.
+	for i := 0; i < 100; i++ {
+		_, err = s.frontendClient.StartWorkflowExecution(ctx, &workflowservice.StartWorkflowExecutionRequest{
+			RequestId:    uuid.New(),
+			Namespace:    "ns_name_los_angeles",
+			WorkflowId:   "wf_id_" + strconv.Itoa(i),
+			WorkflowType: &commonpb.WorkflowType{Name: "workflowTypeName"},
+			TaskQueue:    &taskqueuepb.TaskQueue{Name: "taskQueueName"},
+		})
+		s.NoError(err)
+	}
+
+	// Terminate some workflow executions.
+	for i := 0; i < 30; i++ {
+		_, err = s.frontendClient.TerminateWorkflowExecution(ctx, &workflowservice.TerminateWorkflowExecutionRequest{
+			Namespace: "ns_name_los_angeles",
+			WorkflowExecution: &commonpb.WorkflowExecution{
+				WorkflowId: "wf_id_" + strconv.Itoa(i),
+			},
+		})
+		s.NoError(err)
+	}
+
+	delResp, err := s.operatorClient.DeleteNamespace(ctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: "ns_name_los_angeles",
+	})
+	s.NoError(err)
+	s.Equal("ns_name_los_angeles-deleted-"+nsID[:5], delResp.GetDeletedNamespace())
+
+	descResp2, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Id: nsID,
+	})
+	s.NoError(err)
+	s.Equal(enumspb.NAMESPACE_STATE_DELETED, descResp2.GetNamespaceInfo().GetState())
+
+	namespaceExistsOp := func() error {
```

Comment:
maybe also check that DescribeWorkflowExecution returns namespace not found?

Reply:
This is actually good idea!